### PR TITLE
decompile unk_02002F08.s

### PR DESCRIPTION
**Pull request by remicalixte**
need https://github.com/pret/pokediamond/pull/411 to be merged first

### CODE REVIEW

File: arm9/src/unk_02002F08.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#include "global.h"
+#include "unk_02002F08.h"
+#include "font.h"
+#include "gf_gfx_loader.h"
+#include "string16.h"
+#include "unk_0201B8B8.h"
+
+THUMB_FUNC int FUN_02002F08(u32 param0, struct String *str, u32 param2)
```
Comment (red031000) [inline arm9/src/unk_02002F08.c:8]: does this match with s32?

File: arm9/src/unk_02002F08.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+THUMB_FUNC int FUN_02002F9C(u32 param0, struct String *str)
+{
+    GF_ASSERT(UNK_02106FC8->unk94[param0] != NULL);
+
+    return StringGetWidth_SingleLine_HandleClearToControlCode(UNK_02106FC8->unk94[param0], String_c_str(str));
+}
+
+
```
Comment (red031000) [inline arm9/src/unk_02002F08.c:62]: spacing

File: include/unk_02002F08.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#ifndef POKEDIAMON_UNK_02002F08_H
```
Comment (red031000) [inline include/unk_02002F08.h:1]: POKEDIAMON

File: arm9/src/unk_02002F08.c [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    return r5;
+}
+
+THUMB_FUNC u32 FUN_02002F90(struct String *str)
+{
+    return FUN_02002F58(String_c_str(str));
+}
+
+THUMB_FUNC int FUN_02002F9C(u32 param0, struct String *str)
```
Comment (red031000) [inline arm9/src/unk_02002F08.c:55]: s32?

**Review by red031000 — CHANGES_REQUESTED** [review]

File: include/unk_02002F08.h (hunk 3 above)
Comment (remicalixte) [reply include/unk_02002F08.h]: lol

File: arm9/src/unk_02002F08.c (hunk 1 above)
Comment (remicalixte) [reply arm9/src/unk_02002F08.c:8]: Yes but `StringGetWidth` and `StringGetWidth_SingleLine_HandleClearToControlCode` return an `int` so I used the same return type

Comment (red031000) [reply arm9/src/unk_02002F08.c:8]: do they match with s32s?

Comment (remicalixte) [reply arm9/src/unk_02002F08.c]: Yeah let's just change everything

**Review by red031000 — APPROVED** [review]

**Pull request merged**